mm2src/coins/rpc_command/consolidate_utxos.rs

```diff
@@ -0,0 +1,80 @@
+use common::HttpStatusCode;
+use derive_more::Display;
+use http::StatusCode;
+use mm2_core::mm_ctx::MmArc;
+use mm2_err_handle::prelude::{MmResult, MmResultExt};
+
+use crate::{
+    hd_wallet::HDWalletOps,
+    lp_coinfind_or_err,
+    utxo::{output_script, utxo_builder::merge_utxos},
+    CoinFindError, DerivationMethod, MmCoinEnum,
+};
+
+#[derive(Deserialize)]
+pub struct ConsolidateUtxoRequest {
+    coin: String,
+}
+
+#[derive(Serialize)]
+pub struct ConsolidateUtxoResponse {
+    tx_hash: String,
+}
+
+#[derive(Serialize, Display, SerializeErrorType)]
+#[serde(tag = "error_type", content = "error_data")]
+pub enum ConsolidateUtxoError {
+    NoSuchCoin(String),
+    NotSupportedCoin(String),
+    InvalidAddress(String),
+    MergeError(String),
```

Comment:
Can you make this more typed, like `{ reason: String, coin: String }` instead of `(String)s`? It's hard to understand what is stored inside `String`s (e.g., we put ticker to `NotSupportedCoin` and entire error to `NoSuchCoin` but that's only clear when we read the entire module).